File: src/cli/XtAttributes.cs
using System.Runtime.InteropServices;

namespace Xt
{
    [StructLayout(LayoutKind.Sequential)]
    public struct XtAttributes
    {
        public int size;
        int _isFloat;
        int _isSigned;        
        public bool isFloat { get => _isFloat != 0; set => _isFloat = value ? 0 : 1; }
        public bool isSigned { get => _isSigned != 0; set => _isSigned = value ? 0 : 1; }

        public override string ToString() => XtPrint.AttributesToString(this);
    }
}

File: src/java/com/xtaudio/xt/XtAttributes.java
package com.xtaudio.xt;

import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

public class XtAttributes extends Structure {

    public static class ByValue extends XtAttributes implements Structure.ByValue {}
    
    public int size;
    public boolean isFloat;
    public boolean isSigned;

    @Override
    public String toString() {
        return XtPrint.attributesToString(this);
    }

    @Override
    protected List getFieldOrder() {
        return Arrays.asList("size", "isFloat", "isSigned");
    }
}